onnxruntime/core/providers/cuda/cuda_execution_provider.cc

```diff
@@ -545,9 +563,12 @@ static void RegisterCudaKernels(KernelRegistry& kernel_registry) {
       BuildKernelCreateInfo<ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 7, 9, float, Gemm)>,
       BuildKernelCreateInfo<ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 7, 9, double, Gemm)>,
       BuildKernelCreateInfo<ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 7, 9, MLFloat16, Gemm)>,
-      BuildKernelCreateInfo<ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 1, float, MatMul)>,
-      BuildKernelCreateInfo<ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 1, double, MatMul)>,
-      BuildKernelCreateInfo<ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 1, MLFloat16, MatMul)>,
+      BuildKernelCreateInfo<ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 1, 8, float, MatMul)>,
```

Comment:
There are opset 10 models right now. Merging 1,9 would make it not able to run MatMul ops in opset 10 models.

---
In reply to: [287541933](https://github.com/microsoft/onnxruntime/pull/1106#discussion_r287541933) [](ancestors = 287541933)

Reply:
> There are opset 10 models right now. Merging 1,9 would make it not able to run MatMul ops in opset 10 models.
> 
> In reply to: [287541933](https://github.com/microsoft/onnxruntime/pull/1106#discussion_r287541933) [](ancestors = 287541933)

I think there is a bug for model since version x? If 9 could support 10, why the original setting (1) could not support 9?